src/store/modules/sidebar/sidebar.spec.ts

```diff
@@ -1,15 +1,26 @@
-import { createStore } from 'vuex'
-
-import { storeConfig, State } from '../../index'
+import sidebarModule from './sidebar'
 
 describe('sidebar module', () => {
   describe('actions', () => {
     it('tests getInsights action', async () => {
-      const store = createStore<State>(storeConfig)
+      const rootState: any = {
+        selectedMesh: 'default',
+      }
+      const state: any = sidebarModule.state()
+
+      function commit(type: string, status: any): void {
+        sidebarModule.mutations[type](state, status)
+      }
+
+      function dispatch(type: string): void {
+        // @ts-ignore I can’t be bothered to battle Vuex’s loose types right now.
+        return sidebarModule.actions[type]({ commit, state, rootState })
+      }
 
-      await store.dispatch('sidebar/getInsights')
+      // @ts-ignore I can’t be bothered to battle Vuex’s loose types right now.
```

Comment:
should we create an issue for this?

Reply:
We can’t really fix this in a fashion that wouldn’t be more annoying than leaving this (also annoying) comment. I tried manually setting a narrower type for the actions that only accepts action _handlers_ and not action _objects_ (this union is the cause of the TypeScript error I’m suppressing here), but then the `this` reference of the internal Vuex type structures breaks and I don’t know how to fix it. I tried a few things and in the end it was the least amount of friction to just `@ts-ignore` it.

Without exception all our actions are functions and not plain objects so while this is a true type error, the code path isn’t actually broken.